basyx.aasxfileserver/basyx.aasxfileserver-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/aasxfileserver/InMemoryAasxFileServer.java

```diff
@@ -0,0 +1,139 @@
+	private AtomicInteger packageId = new AtomicInteger(0);
+
+	/**
+	 * Creates the InMemoryAasxFileServer
+	 * @author chaithra
+	 */	
+	public InMemoryAasxFileServer() {}	
+	
+	@Override
+	public Collection<PackageDescription> getAllAASXPackageIds() {
+
+		return packageMap.values().stream()
+				.map(aasxPackage -> aasxPackage.getPackageDescription())
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public InputStream getAASXByPackageId(String packageId) throws ElementDoesNotExistException {
+		throwIfAasxFileIdDoesNotExist(packageId);
+
+		return packageMap.get(packageId).getPackagesBody().getFile();
+	}
+
+	@Override
+	public void updateAASXByPackageId(String packageId, List<String> aasIds, InputStream file, String filename)
+			throws ElementDoesNotExistException {
+		
+		throwIfAasxFileIdDoesNotExist(packageId);
+
+		updateAASXPackage(packageId, aasIds, file, filename);
+	}	
+
+	@Override
+	public PackageDescription createAASXPackage(List<String> aasIds, InputStream file, String fileName)
+			throws CollidingIdentifierException {
+		
+		String newpackageId = String.valueOf(packageId.incrementAndGet());		
+		PackageDescription packageDescription = createPackageDescription(aasIds, newpackageId);		
+		createPackage(aasIds, file, fileName, newpackageId, packageDescription);
+
+		return packageDescription;
+	}	
+
+	@Override
+	public void deleteAASXPackageById(String packageId) throws ElementDoesNotExistException {
+		throwIfAasxFileIdDoesNotExist(packageId);
+
+		packageMap.remove(packageId);
+	}
+
+	public Integer increment(Integer n) {
+		return n++;
+	}
+	
+	private PackageDescription createPackageDescription(List<String> aasIds, String newpackageId) {		
+		PackageDescription packageDescription = new PackageDescription();
+		packageDescription.packageId(newpackageId);
+		packageDescription.aasIds(aasIds);
+		
+		return packageDescription;
+	}
+	
+	private PackagesBody createPackagesBody(List<String> aasIds, InputStream file, String fileName) {		
+		PackagesBody packagesBody = new PackagesBody();
+		packagesBody.aasIds(aasIds);
+		packagesBody.file(file);
+		packagesBody.fileName(fileName);
+		
+		return packagesBody;
+	}	
+	
+	private void createPackage(List<String> aasIds, InputStream file, String fileName, String newpackageId, PackageDescription packageDescription) {
+		PackagesBody packagesBody = createPackagesBody(aasIds, file, fileName);		
+		Package aasxPackage = new Package(newpackageId, packageDescription, packagesBody);		
+		packageMap.put(newpackageId, aasxPackage);
+	}
+	
+	private void updateAASXPackage(String packageId, List<String> aasIds, InputStream file, String filename) {
+		Package aasxPackage = this.packageMap.get(packageId);		
+		aasxPackage.getPackagesBody().setAasIds(aasIds);
+		aasxPackage.getPackagesBody().setFileName(filename);		
+		aasxPackage.getPackagesBody().setFile(file);		
+		aasxPackage.getPackageDescription().setAasIds(aasIds);
+	}	
+	
+	private void throwIfAasxFileIdDoesNotExist(String id) {
```

Comment:
Please rename it to throwIfAasxPackageIdDoesNotExist

Reply:
done